File: conformance/tests/multi-method/multi-method.tsop.go
package contract

import "tsop"

type MultiMethod struct {
	tsop.SmartContract
	Owner  tsop.PubKey `tsop:"readonly"`
	Backup tsop.PubKey `tsop:"readonly"`
}

func (c *MultiMethod) computeThreshold(a tsop.Int, b tsop.Int) tsop.Int {
	return a*b + 1
}

func (c *MultiMethod) SpendWithOwner(sig tsop.Sig, amount tsop.Int) {
	threshold := c.computeThreshold(amount, 2)
	tsop.Assert(threshold > 10)
	tsop.Assert(tsop.CheckSig(sig, c.Owner))
}

func (c *MultiMethod) SpendWithBackup(sig tsop.Sig) {
	tsop.Assert(tsop.CheckSig(sig, c.Backup))
}

File: conformance/tests/multi-method/multi-method.tsop.rs
use tsop::prelude::*;

#[tsop::contract]
struct MultiMethod {
    #[readonly]
    owner: PubKey,
    #[readonly]
    backup: PubKey,
}

#[tsop::methods(MultiMethod)]
impl MultiMethod {
    fn compute_threshold(&self, a: Int, b: Int) -> Int {
        a * b + 1
    }

    #[public]
    fn spend_with_owner(&self, sig: Sig, amount: Int) {
        let threshold = self.compute_threshold(amount, 2);
        assert!(threshold > 10);
        assert!(check_sig(sig, self.owner));
    }

    #[public]
    fn spend_with_backup(&self, sig: Sig) {
        assert!(check_sig(sig, self.backup));
    }
}
